airsonic-main/src/test/java/org/airsonic/player/api/jukebox/AbstractAirsonicRestApiJukeboxIntTest.java

```diff
@@ -46,16 +47,19 @@
 
     @TestConfiguration
     static class Config {
-        private static class SpiedPlayerDaoPlayQueueFactory extends PlayerDaoPlayQueueFactory {
-            @Override
-            public PlayQueue createPlayQueue() {
-                return spy(super.createPlayQueue());
-            }
-        }
-
         @Bean
-        public PlayerDaoPlayQueueFactory playerDaoPlayQueueFactory() {
-            return new SpiedPlayerDaoPlayQueueFactory();
+        public BeanPostProcessor convertToSpy() {
+            return new BeanPostProcessor() {
+                @Override
+                public Object postProcessAfterInitialization(Object bean, String beanName) {
+                    if (bean instanceof PlayerDaoPlayQueueFactory) {
+                        PlayerDaoPlayQueueFactory temp = (PlayerDaoPlayQueueFactory) spy(bean);
+                        doReturn(spy(temp.createPlayQueue())).when(temp).createPlayQueue();
+                        bean = temp;
+                    }
+                    return bean;
+                }
```

Comment:
Implementation note. I chose to implement it this way so we didn't have to enable bean overriding, so it's easier to spot when there are multiple instantiations of the same bean.

An alternate implementation is to of course turn on bean overriding for tests.

Reply:
Turning on bean overriding did not seem to be very future-proof given that they pretty much deprecated it. I prefer this solution!